### PR TITLE
MM-48401 : Convert "sidebar_category_sorting_menu"  to functional components and replace with compass icons

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Migrated `sidebar_category_sorting_menu` to functional component. Replaced the icons to compass-icons.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48401

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: sass/layout/_sidebar-left.scss [hunk 1]
```diff
@@ -373,6 +373,10 @@ $sidebarOpacityAnimationDuration: 0.15s;
                 font-size: 16px;
                 line-height: 16px;
             }
+
+            svg {
+                color: rgba(var(--center-channel-color-rgb), 0.72);
```
Comment (neallred) [inline sass/layout/_sidebar-left.scss:378]: :+1:

**Review by neallred — APPROVED** [review]
LGTM besides failing snapshots. Thanks, @M-ZubairAhmed

**Review by m1lt0n — APPROVED** [review]
Looks good! Quick question: I see no tests are affected by this. Does the component have tests? If not, perhaps having a minimal test suite to verify it works as before would be great

**Comment by M-ZubairAhmed** [comment]
2 tests in `tests/integration/messaging/save_post_spec.js` failed in [e2e report](https://automation-dashboard.vercel.app/cycles/f377ff7a-98a2-42d1-9d32-d867c82d3b87). Ran them both locally, they are a little bit flaky . Not related to this PR.

**Pull request merged**